common.c

```diff
@@ -52,8 +57,22 @@
  * calls into your tapped version of getuid and we
  * get into an infinite loop.
  *
+ * g_enable_tracing is used only for the main thread,
+ * if we see ourselves being called from a different thread
+ * we will initialized the key g_enable_tracing_key
+ * and will use that to store a per-thread tracing
+ * enabled/disabled setting. I used to declare
+ * g_enable_tracing with the __thread modifier, but in
+ * macOS that will eventually call malloc which would
+ * get us into a loop.
+ *
+ * g_enable_tracing_key logic is backwards, 0 (NULL)
+ * means tracing is enabled. 1 means tracing is disabled.
+ *
```

Comment:
It's a tricky one. Is real_malloc not an option? it calls other functions I suppose.
Perhaps an enum
```
enum {
        TRACING_ENABLED = 1,
        TRACING_DISABLED
}
```
and use that for the g_tracing_enabled too. That's probably what you had in mind anyway :)